src/components/AddUser/AddUser.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, FormEvent, FC } from 'react';
+import { Todo } from '../../types/types';
+import { getUserById } from '../../utils/getUserByID';
+import usersFromServer from '../../api/users';
+
+interface AddUserProps {
+  onAdd: (newTodo: Todo) => void;
+  todos: Todo[]; // We need todos to calculate max id
+}
+
+export const AddUser: FC<AddUserProps> = ({ onAdd, todos }) => {
+  const [newTitle, setNewTitle] = useState('');
+  const [newUserId, setNewUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const getNextId = () => {
+    if (todos.length === 0) {
+      return 1;
+    }
+
+    const maxId = Math.max(...todos.map(todo => todo.id));
+
+    return maxId + 1;
+  };
+
+  function handleFormReset() {
+    setNewTitle('');
+    setNewUserId(0);
+    setTitleError(false);
+    setUserError(false);
+  }
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    let hasError = false;
+
+    if (!newTitle) {
+      setTitleError(true);
+      hasError = true;
+    }
+
+    if (!newUserId) {
+      setUserError(true);
+      hasError = true;
+    }
+
+    if (hasError) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getNextId(),
+      title: newTitle,
+      completed: false,
+      userId: newUserId,
+      user: getUserById(newUserId),
+    };
+
+    onAdd(newTodo);
+    handleFormReset();
+  }
```

Comment:
[CODE KNOWLEDGE] - While handling form values, trigger submit logic using `onSubmit` prop of form, instead of `onClick` submit button. You've done this correctly, good job!